### PR TITLE
extract logging

### PR DESCRIPTION
**Pull request by whyrusleeping**
Pulls logging out into `ipfs/go-log` and also consolidates all logging to our 'eventlog' library, on top of logrus.

License: MIT
Signed-off-by: Jeromy <jeromyj@gmail.com>

### CODE REVIEW

**Comment by jbenet** [comment]
LGTM

File: vendor/QmXJkcEXB6C9h6Ytb6rrUTFU56Ro62zxgrbxTT3dgjQGA8/go-log/oldlog.go [hunk 1]
```diff
@@ -1,6 +1,7 @@
-package util
+package log
 
 import (
+	"errors"
 	"os"
 
 	"github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/Sirupsen/logrus"
```
Comment (rht) [inline vendor/QmXJkcEXB6C9h6Ytb6rrUTFU56Ro62zxgrbxTT3dgjQGA8/go-log/oldlog.go]: Should this be moved to go-log's Godeps and that `github.com/ipfs/go-ipfs/thirdparty/pollEndpoint` uses go-log as well?

Comment (jbenet) [inline vendor/QmXJkcEXB6C9h6Ytb6rrUTFU56Ro62zxgrbxTT3dgjQGA8/go-log/oldlog.go]: s/Godeps/gxdeps/ but yes.

Comment (rht) [inline vendor/QmXJkcEXB6C9h6Ytb6rrUTFU56Ro62zxgrbxTT3dgjQGA8/go-log/oldlog.go]: was wondering of that as well

**Pull request merged**